backend/chart.go

```diff
@@ -119,12 +121,21 @@ func (backend *Backend) ChartData() (*Chart, error) {
 
 	fiat := backend.Config().AppConfig().Backend.MainFiat
 	// Chart data until this point in time.
-	until := backend.RatesUpdater().HistoryLatestTimestampAll(backend.allCoinCodes(), fiat)
+	until, timestamps := backend.RatesUpdater().HistoryLatestTimestampAll(backend.allCoinCodes(), fiat)
 	if until.IsZero() {
 		chartDataMissing = true
 		backend.log.Info("ChartDataMissing, until is zero")
 	}
-	isUpToDate := time.Since(until) < 2*time.Hour
+	upToDateLimit := 2 * time.Hour
+	isUpToDate := time.Since(until) < upToDateLimit
+
+	unixTimestamps := make(map[string]int64)
+	for coin, timestamp := range timestamps {
+		if time.Since(timestamp) >= upToDateLimit {
+			// js Date is based on milliseconds.
+			unixTimestamps[coin] = timestamp.UnixMilli()
```

Comment:
if i read the frontend code correctly you only really display `until` there (smallest of the timestamps), so making this `unixTimestamps` map is not needed?

imo it would be much clearer to pass a `outdated { CoinsNames []string, timestamp time.Time }` struct.

Reply:
Yes, because we initially decided to display the last rates available for each coin/fiat pair. This is not used anymore now and can be removed if we don't plan to add this information anymore.